Source/Eto/Forms/Controls/DocumentControl.cs

```diff
@@ -0,0 +1,371 @@
+		{
+			add { Properties.AddEvent(SelectedIndexChangedEvent, value); }
+			remove { Properties.RemoveEvent(SelectedIndexChangedEvent, value); }
+		}
+
+		/// <summary>
+		/// Raises the <see cref="PageClosed"/> event.
+		/// </summary>
+		/// <param name="e">Event arguments.</param>
+		protected virtual void OnPageClosed(DocumentPageEventArgs e)
+		{
+			Properties.TriggerEvent(PageClosedEvent, this, e);
+			var page = SelectedPage;
+			if (page != null)
+				page.TriggerClose(e);
+		}
+
+		/// <summary>
+		/// Raises the <see cref="SelectedIndexChanged"/> event.
+		/// </summary>
+		/// <param name="e">Event arguments.</param>
+		protected virtual void OnSelectedIndexChanged(EventArgs e)
+		{
+			Properties.TriggerEvent(SelectedIndexChangedEvent, this, e);
+			var page = SelectedPage;
+			if (page != null)
+				page.TriggerClick(e);
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="DocumentControl"/> class.
+		/// </summary>
+		public DocumentControl()
+		{
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="DocumentControl"/> class with the specified handler.
+		/// </summary>
+		/// <param name="handler">Handler for the implementation of the tab control.</param>
+		protected DocumentControl(IHandler handler)
+			: base(handler)
+		{
+		}
+
+		/// <summary>
+		/// Gets or sets the index of the selected tab.
+		/// </summary>
+		/// <value>The index of the selected tab.</value>
+		public int SelectedIndex
+		{
+			get { return Handler.SelectedIndex; }
+			set { Handler.SelectedIndex = value; }
+		}
+
+		/// <summary>
+		/// Gets or sets the currently selected page.
+		/// </summary>
+		/// <value>The selected page.</value>
+		public DocumentPage SelectedPage
+		{
+			get { return SelectedIndex < 0 ? null : Pages[SelectedIndex]; }
+			set { SelectedIndex = pages.IndexOf(value); }
+		}
+
+		/// <summary>
+		/// Gets the collection of tab pages.
+		/// </summary>
+		/// <value>The pages.</value>
+		public IList<DocumentPage> Pages
+		{
+			get { return pages ?? (pages = new DocumentPageCollection(this.Handler)); }
+		}
+
+		/// <summary>
+		/// Remove the specified child from the container.
+		/// </summary>
+		/// <param name="child">Child to remove.</param>
+		public override void Remove(Control child)
+		{
+			var page = child as DocumentPage;
+			if (page != null)
+			{
+				Pages.Remove(page);
+			}
+		}
+
+		/// <summary>
+		/// Gets the binding for the <see cref="SelectedIndex"/> property.
+		/// </summary>
+		/// <value>The selected index binding.</value>
+		public BindableBinding<DocumentControl, int> SelectedIndexBinding
+		{
+			get
+			{
+				return new BindableBinding<DocumentControl, int>(
+					this,
+					c => c.SelectedIndex,
+					(c, v) => c.SelectedIndex = v,
+					(c, h) => c.SelectedIndexChanged += h,
+					(c, h) => c.SelectedIndexChanged -= h
+				);
+			}
+		}
+
+		class DocumentPageCollection : IList<DocumentPage>
```

Comment:
Hm, any reason this isn't a `Collection<DocumentPage>`?

It seems that it might be better to use, as it'd eliminate the need for `IHandler.GetPage()`, as well as the need for most of the IList<T> implementation.